### PR TITLE
Add RoomMessageEvent

### PR DESCRIPTION
**Pull request by Shiranuit**
## What does this PR do ?

- Refactor event names
- Refactor message handling logic
- Add RoomMessageEvent
- Add UnhandledExceptionEvent

### CODE REVIEW

File: src/main/kotlin/io/kuzzle/sdk/Kuzzle.kt [hunk 1]
```diff
@@ -90,28 +92,26 @@ open class Kuzzle {
...
-            protocol.trigger(UnhandledResponseEvent(message))
-            return
+        if (response.error?.id == "security.token.expired") {
+            protocol.trigger(TokenExpiredEvent())
         }
 
-        if (response.error == null) {
-            queries[requestId]?.complete(response)
-            queries.remove(requestId)
+        if (queries.size == 0 || requestId == null || queries[requestId] == null) {
```
Comment (OlivierCavadenti) [inline src/main/kotlin/io/kuzzle/sdk/Kuzzle.kt:99]: queries.size == 0 => queries.isEmpty()

File: src/main/kotlin/io/kuzzle/sdk/controllers/RealtimeController.kt [hunk 2]
```diff
@@ -25,28 +24,23 @@ class RealtimeController(kuzzle: Kuzzle) : BaseController(kuzzle) {
...
-                val subs: ArrayList<Subscription>? = currentSubscriptions[response.room]
-                if (subs != null) {
-                    val instanceId = sdkInstanceId
-                    subs.forEach {
-                        if (instanceId == kuzzle.instanceId && it.subscribeToSelf || instanceId != kuzzle.instanceId) {
-                            it.handler(response)
-                        }
+            val subs: ArrayList<Subscription>? = currentSubscriptions[response.room]
+            if (subs != null) {
+                val instanceId = sdkInstanceId
```
Comment (OlivierCavadenti) [inline src/main/kotlin/io/kuzzle/sdk/controllers/RealtimeController.kt:36]: why create new variable ?

Comment (Shiranuit) [reply src/main/kotlin/io/kuzzle/sdk/controllers/RealtimeController.kt:36]: I didn't write this code, I just rearanged it

**Comment by OlivierCavadenti** [comment]
PR now have my commits, it's desired ?

**Pull request merged**